Mechanism function order code generation bug in CUDA

<!-- Please add relevant labels to categorize your bug. -->

**Describe the bug**
<!-- A clear and concise description of what the bug is, steps to reproduce, expected behavior, etc. -->

When [this NMODL file](https://github.com/dbbs-lab/catalogue/blob/main/dbbs_catalogue/mod/Kca3_1.mod) is compiled for CUDA, the `vdep` function is declared after the `rate` function which uses the former, causing the following warning and error:

```
/users/bp000347/arbor-gpu/arbor/build/mechanisms/generated/default/Kca3_1_gpu.cu(67): error: identifier "vdep" is undefined

/users/bp000347/arbor-gpu/arbor/build/mechanisms/generated/default/Kca3_1_gpu.cu(75): warning: function "arb::default_catalogue::<unnamed>::vdep" was declared but never referenced

1 error detected in the compilation of "/users/bp000347/arbor-gpu/arbor/build/mechanisms/generated/default/Kca3_1_gpu.cu".
make[2]: *** [arbor/CMakeFiles/arbor.dir/build.make:3053: arbor/CMakeFiles/arbor.dir/__/mechanisms/generated/default/Kca3_1_gpu.cu.o] Error 1
```

--- a/modcc/printer/gpuprinter.cpp
+++ b/modcc/printer/gpuprinter.cpp
@@ -186,16 +186,26 @@ std::string emit_gpu_cu_source(const Module& module_, const printer_options& opt
         << "using ::arb::gpu::max;\n\n";
 
     // Procedures as __device__ functions.
-    auto emit_procedure_kernel = [&] (ProcedureExpression* e) {
+    auto emit_procedure_proto = [&] (ProcedureExpression* e) {
         out << fmt::format("__device__\n"
                            "void {}(arb_mechanism_ppack params_, int tid_",
                            e->name());
         for(auto& arg: e->args()) out << ", arb_value_type " << arg->is_argument()->name();
-        out << ") {\n" << indent
+        out << ")";
+    };
+
+    auto emit_procedure_kernel = [&] (ProcedureExpression* e) {
+        emit_procedure_proto(e);
+        out << " {\n" << indent
             << "PPACK_IFACE_BLOCK;\n"
             << cuprint(e->body())
             << popindent << "}\n\n";
     };
+
+    for (auto& p: module_normal_procedures(module_)) {
+        emit_procedure_proto(p);
+        out << ";\n\n";
+    }
 
     for (auto& p: module_normal_procedures(module_)) {
         emit_procedure_kernel(p);

